File: Java/src/Y2020/Day1/ReportRepair.java
package Y2020.Day1;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class ReportRepair {
    private List<Integer> data = new ArrayList<>();

    public ReportRepair() throws FileNotFoundException {
        Scanner sc = new Scanner(new File("./src/Y2020/Day1/input.txt"));
        while (sc.hasNextInt())
            this.data.add(sc.nextInt());

        this.part1();
        this.part2();
    }

    private void part1() {
        outer: for (int i = 0; i < this.data.size(); i++) {
            for (int j = i; j < this.data.size(); j++) {
                int ii = this.data.get(i);
                int jj = this.data.get(j);
                if (ii + jj == 2020) {
                    System.out.println(ii * jj);
                    break outer;
                }
            }
        }
    }

    private void part2() {
        outer: for (int i = 0; i < this.data.size(); i++) {
            for (int j = i; j < this.data.size(); j++) {
                for (int k = j; k < this.data.size(); k++) {
                    int ii = this.data.get(i);
                    int jj = this.data.get(j);
                    int kk = this.data.get(k);
                    if (ii + jj + kk == 2020) {
                        System.out.println(ii * jj * kk);
                        break outer;
                    }
                }
            }
        }
    }

    public static void main(String[] args) throws FileNotFoundException {
        new ReportRepair();
    }
}

File: Python/Y2020/Day1/ReportRepair.py
data = []
with open("input.txt", "r") as file:
    for line in file:
        data.append(int(line))

part1, part2 = None, None
for i in range(len(data)):
    for j in range(i, len(data)):
        if data[i] + data[j] == 2020:
            part1 = data[i] * data[j]
        for k in range(j, len(data)):
            if data[i] + data[j] + data[k] == 2020:
                part2 = data[i] * data[j] * data[k]

print(f"Part 1: {part1}")
print(f"Part 2: {part2}")
